lib/request.js

```diff
@@ -138,6 +140,171 @@ function buildRequestWithTrustProxy (R, trustProxy) {
   return _Request
 }
 
+function isValidHostname (hostname) {
+  if (!hostname) {
+    return false
+  }
+  if (net.isIP(hostname)) {
+    return true
+  }
+  return /^(?=.{1,253}$)(?!-)[A-Za-z0-9-]+(\.[A-Za-z0-9-]+)*(?<!-)$/.test(hostname)
```

Comment:
Whats this?

Reply:
It makes sure that the hostname conforms to the rules of either IPv4, IPv6 or the rules defined at https://en.wikipedia.org/wiki/Hostname#Syntax

Reply:
I am against adding complicated regular expressions. _Especially_ if they are not fully documented with a doc block explaining each component and examples of what it matches.